### DISCUSSION TITLE
expansion after dollar sign

### DISCUSSION THREAD

**Discussion by jonas-frey** [draivin/hsnips#90] [Q&A] [answered]
I'm using hsnips to write latex and markdown, and I would like to be able to enter a shortcut immediately after a dollar sign starting math mode. Right now I have to leave a space after the dollar sign for the snippet to trigger. Is it possible to have snippets trigger after dollar signs as well?

**Comment by yfzhao20** [comment]
Can you give me an example？ I will try solving that.

**Reply by jonas-frey** [reply]
For example I have a snippet
```
snippet CC A
\mathbb C
endsnippet
```
and when I write `$CC` I want it to expand to $\mathbb C$.

In markdown (like in latex), the dollar sign is used to enter (and exit) 'math mode', but markdown is stricter than latex and only enters math mode if there's no space after the dollar sign. Therefore the easy workaround of typing a space after the dollar sign doesn't work.

Actually it would also be nice to have snippet expansion after other symbols like `{` . Would it be possible to have a config option in which to specify after which symbols snippets expand?

**Reply by yfzhao20** [reply]
Do you mean ... this ?
![1111](https://user-images.githubusercontent.com/80528249/137827555-e87aa664-7b24-4f21-a9ef-47b62054f691.gif)

**Reply by jonas-frey** [reply]
Yes, but the thing is that I want to use CC not only after a dollar sign -- ie at the beginning of math mode--, but anywhere in math mode.

So the option that you're suggesting would require to create a second version for every shortcut that I want to be able to use after a dollar sign. And if I also want to be able to use it also after `{` then I have to create a third version.

**Reply by yfzhao20** [reply]
I think so... but there are not sufficient options provided by hsnips. And maybe I will set it twice:
```lua

snippet $CC A
$\mathbb{C}$
endsnippet

context math(context)
snippet CC A
\mathbb{C}
endsnippet

```

**Reply by draivin** [reply, marked as answer]
It seems that what you need is to add the flag `i` to your snippet, as can be seen in the readme, this enables in-word expansion. So it would be something like
```lua
snippet $CC Ai
$\mathbb{C}$
endsnippet
```

**Reply by jonas-frey** [reply]
Yes, the `i` flag is what I was looking for. 

(Apparently I can't accept this as solution since it's not a top-level answer)
